src/server/NetworkingServer.cpp

```diff
@@ -163,36 +180,53 @@ void NetworkingServer::ComputeGameState() {
         mAcceptingMoveCommands = true;
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
         mAcceptingMoveCommands = false;
-        // update game state, take moves array and simulate next tick given current state and inputs
-        // ... simplest behavior would be to track client locations and apply vectors to them ...
 
-        if (mClientMoves.size() == 0) {
-            continue;
-        }
-        std::cout << "game state thread: dumping client moves for tick " << mCurrTick << std::endl;
-        for (const auto iter : mClientMoves) {
-            sf::Uint32 client_id = iter.first;
-            float* move_vector = iter.second;
-            if (mClientPositions.find(client_id) != mClientPositions.end()) {
-                mClientPositions[client_id][0] += move_vector[0]*5.f; // x component
-                mClientPositions[client_id][1] += move_vector[1]*5.f; // y component
-            }
-            std::cout << iter.first << " " << iter.second[0] << " " << iter.second[1] << std::endl;
-        }
-        std::cout << "game state thread: dumping client positions for tick " << mCurrTick << std::endl;
-        for (const auto iter : mClientPositions) {
-            std::cout << iter.first << " " << iter.second[0] << " " << iter.second[1] << std::endl;
-        }
         // window is closed, time to reset. May want to use fix sized array as an optimization.
         // It's a bit mean to disguard old commands, maybe there should be windows for multiple ticks?
-        mClientMoves.clear();
         mCurrTick++;
+        // This sleep is needed to give the GameController time to fetch the client directions in getClientDirections
+        // TODO: Not sure what this sleep value should be...
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
     }
 }
 
 void NetworkingServer::DeleteClient(sf::TcpSocket* client, sf::SocketSelector selector) {
     mClientMoves.erase(mClientSocketsToIds[client]);
-    mClientPositions.erase(mClientSocketsToIds[client]);
     selector.remove(*client);
+    mClientSocketsToIds.erase(client);
     delete client;
 }
+
+std::vector<network_player> NetworkingServer::getNetworkPlayers() {
+    std::vector<network_player> network_players;
+    if (mAcceptingMoveCommands == false) {
+        for(const auto& client_move : mClientMoves) {
+            network_player np = {client_move.first, client_move.second[0], client_move.second[1]};
+            network_players.push_back(np);
+        }
+    }
+    return network_players;
+}
+
+void NetworkingServer::setState(std::vector<Group*> active_groups) {
```

Comment:
eh I rather be as specific as possible. Maybe call this `setNetworkGameObjects`

Reply:
This is meant to be the single function that sets the entire state that needs to be retrieved by clients. Right now we only have groups that are converted to `network_game_objects` but in the future we will also be setting mines, resources, etc.

I don't mind keeping it `setState` so that it indicates how it's meant to be expanded.

Reply:
I'd put that in a comment. I worry that we make assumptions about the future that often don't end up true... because it's the unknowable future. I respect that at some point that's how this might end up, but we're not there and I'd prefer clarity now over one less search and replace later if that makes sense.